File: structural/Java/Bridge.java


interface Fuel {

	public String getType();

}

class Diesel implements Fuel {

	@Override
	public String getType() {
		return "Diesel";
	}

}

class Gasoline implements Fuel {

	@Override
	public String getType() {
		return "Gasoline";
	}

}

interface Car {

	public void showInfo();

}

class Jeep implements Car {

	private Fuel fuel;

	public Jeep(Fuel fuel) {
		this.fuel = fuel;
	}

	@Override
	public void showInfo() {
		System.out.println("\nModel: Jeep");
		System.out.println("Fuel: " + fuel.getType());
	}
}

class Subaru implements Car {

	private Fuel fuel;

	public Subaru(Fuel fuel) {
		this.fuel = fuel;
	}

	@Override
	public void showInfo() {
		System.out.println("\nModel: Subaru");
		System.out.println("Fuel: " + fuel.getType());
	}

}

public class Main {

	public static void main(String[] args) {
		System.out.println(":: Bridge ::\n");

		Fuel diesel = new Diesel();
		Fuel gasoline = new Gasoline();

		Car jeep = new Jeep(diesel);
		Car subaru = new Subaru(gasoline);

		jeep.showInfo();
		subaru.showInfo();
	}

}

File: structural/PHP/Bridge.php
<?php

interface Fuel {

    public function getType();

}

class Diesel implements Fuel {

    public function getType() {
        return 'Diesel';
    }

}

class Gasoline implements Fuel {

    public function getType() {
        return 'Gasoline';
    }

}

interface Car {

    public function showInfo();

}

class Jeep implements Car {

    private $fuel;

    public function __construct($fuel) {
        $this->fuel = $fuel;
    }

    public function showInfo() {
        echo "\nModel: Jeep";
        echo "\nFuel: " . $this->fuel->getType() . "\n";
    }

}

class Subaru implements Car {

    private $fuel;

    public function __construct($fuel) {
        $this->fuel = $fuel;
    }

    public function showInfo() {
        echo "\nModel: Subaru";
        echo "\nFuel: " . $this->fuel->getType() . "\n";
    }

}

echo ":: Bridge ::\n";

$diesel = new Diesel();
$gasoline = new Gasoline();

$jeep = new Jeep($diesel);
$subaru = new Subaru($gasoline);

$jeep->showInfo();
$subaru->showInfo();

?>